gobblin-core/src/main/java/gobblin/http/ResponseHandler.java

```diff
@@ -0,0 +1,10 @@
+package gobblin.http;
+
+/**
+ * An interface to handle a response
+ *
+ * @param <RP> type of response
+ */
+public interface ResponseHandler<RP> {
```

Comment:
That works but it's not necessary. This interface is only used in http domain. Currently, `ResponseStatus` only has the status code. It can be subclassed as any other type, which can do whatever a template does. Returning `ResponseStatus` facilitates add common logic after handling the response in a super class and reduces the number of generics.